### PR TITLE
Add check-cluster-up script for KinD providers

### PR DESCRIPTION
**Pull request by ormergi**
Currently we dont run Kubevirt conformance tests on SR-IOV check up lane
This PR adds check-cluster-up script for KinD based providers, similar to what we have on check-provision lanes.

Basically it will check that the cluster is healthy, dump the nodes network configurations
and run conformance tests.

This script will be used on SR-IOV prow job command in order to prevent bad changes to SR-IOV provider.

Signed-off-by: Or Mergi <ormergi@redhat.com>

### CODE REVIEW

**Comment by oshoval** [comment]
> This PR check-cluster-up script for KinD based providers, similar to what we have on check-provision lanes.

Missing the word `adds`

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [inline cluster-up/cluster/kind/check-cluster-up.sh:38]: please use `get nodes` here,
and also there is double space before

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [inline cluster-up/cluster/kind/check-cluster-up.sh:57]: please add `-s` (all over)

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [inline cluster-up/cluster/kind/check-cluster-up.sh:60]: maybe worth to assert that kubevirt is not deployed instead ?
i do understand it gives freedom to install specific kubevirt, so up to you

btw usually on kubevirtci the sonobuoy conformance are without kubevirt, right?

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [inline cluster-up/cluster/kind/check-cluster-up.sh:64]: maybe should be part of the if, else it means that the kubevirt is deployed but here you are checking that its also Available, again up to you

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [inline cluster-up/cluster/kind/check-cluster-up.sh:68]: `="$SONOBUOY_EXTRA_ARGS ..."` first i think (also in the 2nd place)

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [inline cluster-up/cluster/kind/check-cluster-up.sh:71]: maybe the `0:9` should be in the previous line ?

**Review by oshoval — COMMENTED** [review]
Thanks

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh:60]: > maybe worth to assert that kubevirt is not deployed instead ?
> i do understand it gives freedom to install specific kubevirt, so up to you

In case its not deployed it wont get into the if block

> btw usually on kubevirtci the sonobuoy conformance are without kubevirt, right?

Currently we run Kubevirt conformance as well on all check-provision lanes

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh:64]: This check is not part of the if in order to prevent running tests in case not all kubevirt components are ready.

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh:68]: In general the order doesn't matter, I placed `SONOBUOY_EXTRA_ARGS` at the end so it will be easier to understand what we just added.

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh:71]: you mean adding another line?
`commit=${commit:0:9}`

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh]: Done

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh]: Done

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [reply cluster-up/cluster/kind/check-cluster-up.sh:68]: of course it doesnt matter, but most of the places all over our code are appending, imo it should be the same here

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [reply cluster-up/cluster/kind/check-cluster-up.sh:71]: ye, just so it will be less hidden

Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh]: Done

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh]: Done

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [reply cluster-up/cluster/kind/check-cluster-up.sh:64]: maybe 15 minutes is too much ?
how much other repos use

**Review by oshoval — APPROVED** [review]
Thanks

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [inline cluster-up/cluster/kind/check-cluster-up.sh:39]: some thing nice is to create an array of command strings and then iterate on it , executing it,
wdyt?

**Comment by ormergi** [comment]
Added missing quotes to kubevirt deployed if condition

**Comment by oshoval** [comment]
I think that we need a way to easily run the provision without checking kubevirt as well
maybe a 2nd lane which is optional and does all the same, beside the check-up?

else it will make testing where there are iterations on CI, slower
and will need hacking in order to skip temporarily the check

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh:39]: I agree its nicer, I didnt do it because I wanted to keep the log messages before each command.
Putting it in bash equivalent for map doesn't make things prettier in this case.

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh:64]: on hack/cluster-deploy.sh it check every 5 minutes for 25 minutes
https://github.com/kubevirt/kubevirt/blob/371958b07a340d0ba9e9d1e9a2936411e395a5df/hack/cluster-deploy.sh#L154
From what I saw sometime it takes 2-3 rounds to be ready, rarely more then 10 minutes , so I picked 15 minutes timeout.

**Comment by ormergi** [comment]
> I think that we need a way to easily run the provision without checking kubevirt as well
> maybe a 2nd lane which is optional and does all the same, beside the check-up?
> 
> else it will make testing where there are iterations on CI, slower
> and will need hacking in order to skip temporarily the check

I understand it could slow down iterations on CI, but think about the time it will save to revert a bad change 
after you discovered it only when its land at kubevirt/kubevirt, 
the time you would spend on debugging it.

Instead we will get initial feedback of how a change will affect kubevirt/kubevirt, 
Also it will save the hassle to create secondary PR and test the changes on kubevirt/kubevirt.

For iterating on CI Its pretty straight forward to skip Kubevirt conformance,
all you need to do is override RUN_KUBEVIRT_CONFORMANCE env var (on provider.sh)
I think is simple then having another secondary job that does only check-up (w/o conformance)

File: cluster-up/cluster/kind/check-cluster-up.sh
Comment (oshoval) [reply cluster-up/cluster/kind/check-cluster-up.sh:39]: You can still keep log messages or whatever you desire in the loop isnt it ?
something like this wouldn't be nicer?
```
declare -a arr=("element1" "element2" "element3")
for i in "${arr[@]}"
do
   echo "$i" # exec here instead
done
```

Comment (ormergi) [reply cluster-up/cluster/kind/check-cluster-up.sh:39]: > You can still keep log messages or whatever you desire in the loop isnt it ?

It will print the command we executes and not its descriptions, for example
for `ip a` command, it will print 
`[sriov-worker] ip r` 
instead of:
`network interfaces status:`

It makes the log easier to follow, cosmetics.

**Comment by oshoval** [comment]
the last push is pure rebase right ?

**Comment by ormergi** [comment]
> the last push is pure rebase right ?

Yes

**Review by oshoval — APPROVED** [review]
Thanks

**Review by dhiller — APPROVED** [review]
/approve

Sorry for the delay, this slipped through :-/

**Comment by ormergi** [comment]
/hold 

Wait until https://github.com/kubevirt/kubevirt/pull/6199 is merged as we might need to do some changes to this PR

**Pull request merged**
